Formula/docker-ls.rb

```diff
@@ -0,0 +1,31 @@
+class DockerLs < Formula
+  desc "Tools for browsing and manipulating docker registries."
+  homepage "https://github.com/mayflower/docker-ls"
+  url "https://github.com/mayflower/docker-ls.git",
+    :tag => "v0.3.1",
+    :revision => "d80310976c9707e261e57ebfa9acf4e0b1781460"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+
+    (buildpath/"src/github.com/mayflower/docker-ls").install buildpath.children
+
+    system "go", "generate", "github.com/mayflower/docker-ls/lib"
```

Comment:
Because of the above line we're still not gaining automatic metafiles installation here so we might as well just use the standard format, like [this sort of thing](https://github.com/Homebrew/homebrew-core/blob/58023f24d36598262002a20537c69ab38f517bf4/Formula/goose.rb#L20-L25).

Reply:
I must admit that I don't quite understand. From looking at the source, `install_metafiles` seems to install `README.md` and friends, which is fine --- I can add this. However, I don't understand how goose gets its binaries installed?

Reply:
You want to do this sort of thing:
```diff
diff --git a/Formula/docker-ls.rb b/Formula/docker-ls.rb
index c967a8b660..1509451ae3 100644
--- a/Formula/docker-ls.rb
+++ b/Formula/docker-ls.rb
@@ -9,14 +9,13 @@ class DockerLs < Formula
 
   def install
     ENV["GOPATH"] = buildpath
-
     (buildpath/"src/github.com/mayflower/docker-ls").install buildpath.children
 
-    system "go", "generate", "github.com/mayflower/docker-ls/lib"
-    system "go", "install", "github.com/mayflower/docker-ls/cli/..."
-
-    bin.install buildpath/"bin/docker-ls"
-    bin.install buildpath/"bin/docker-rm"
+    cd "src/github.com/mayflower/docker-ls" do
+      system "go", "build", "-o", bin/"docker-ls", "./cli/docker-ls"
+      system "go", "build", "-o", bin/"docker-rm", "./cli/docker-rm"
+      prefix.install_metafiles
+    end
   end
 
   test do
```

Reply:
Hopefully that's fairly self-explanatory but please feel free to ask otherwise.

Reply:
@DomT4 I have made the changes and updated the PR. Looking at the code again it is quite clear what is going on. Sorry for the late response.